src/sentry/static/sentry/app/utils/apm.tsx

```diff
@@ -9,3 +17,54 @@ export function setTransactionName(name: string) {
     scope.setTag('ui.route', name);
   });
 }
+
+export async function normalizeTransactionName(
+  event: Sentry.Event
+): Promise<Sentry.Event> {
+  if (event.type === 'transaction') {
+    // For JavaScript transactions, translate the transaction name if it exists and doesn't start with /
+    // using the app's react-router routes. If the transaction name doesn't exist, use the window.location.pathname
+    // as the fallback.
+
+    let prevTransactionName = event.transaction;
+
+    if (typeof prevTransactionName === 'string') {
+      if (prevTransactionName.startsWith('/')) {
+        return event;
+      }
+    } else {
+      prevTransactionName = window.location.pathname;
+    }
+
+    const transactionName: string | undefined = await new Promise(function(resolve) {
+      Router.match(
+        {
+          routes: appRoutes,
+          location: createLocation(prevTransactionName),
+        },
+        (error, _redirectLocation, renderProps) => {
+          if (error) {
+            return resolve(undefined);
+          }
+
+          const routePath = getRouteStringFromRoutes(renderProps.routes ?? []);
+          return resolve(routePath);
+        }
+      );
+    });
+
+    if (typeof transactionName === 'string' && transactionName.length) {
+      event.transaction = transactionName;
+
+      if (event.tags) {
+        event.tags['ui.route'] = transactionName;
```

Comment:
Should we tag this event to identify how often this happens?

Reply:
@billyvg do you mean analytics?

Reply:
@dashed Yeah, might be nice to have it as a tag in the event

Reply:
@billyvg is this something you had in mind? https://github.com/getsentry/sentry/pull/18822/commits/58bb0bfe9b9856c6e3e58c7826b3ae2bea3282f2

Reply:
@dashed Yeah, you may be able to simplify to `prev !== transactionName` if you don't need the prev transaction name (since you already have transaction name).

Reply:
@billyvg I'm unsure what you meant.  `prevTransactionName !== transactionName` should be guaranteed to be true in this code path.

Reply:
Ah, then the previous tag should be enough, and if you wanted to know the number of transactions that required normalization you can query for something like `!transaction.rename.before:''`?

Reply:
I think `has:transaction.rename.source` might be the search condition. Since this gets set knowing that normalization is required. And regardless of whether or not the normalization process was successful.